src/BinaryParsers/PEBinary/PEBinary.cs

```diff
@@ -121,5 +93,54 @@ public static bool CanLoadBinary(Uri uri)
             catch (IOException) { return false; }
             catch (UnauthorizedAccessException) { return false; }
         }
+
+        private Pdb LoadPdb()
+        {
+            string pePath = this.PE?.FileName ?? this.TargetUri.LocalPath;
+            if (!TryLoadPdb(pePath, this.symbolPath, this.localSymbolDirectories, this.tracePdbLoad, out Pdb pdb))

```

Comment:
>TryLoadPdb [](http://example.com/codeflow?start=17&length=10)

If you know that a file with an exe or dll extension will fail, why don't you check for that upfront? Having a TryLoadPdb call nested inside this is a clue that you should consider clarifying inputs upfront.

Reply:
normally, the first try points to the binary itself and when that happens, we  try to load the pdb using `loadDataForExe`.

But, even in cases where the dll and pdb are side by side, the `loadDataForExe` can not load, so, i'm forcing loading only the pdb, calling `loadDataFromPdb`.

i understood ur concern, but if we read directly the pdb, it will be easier, but we would change the behavior...

we can talk more later to discuss :)